### PR TITLE
Distinction between max payload and max pax

### PR DESCRIPTION
**Pull request by ngomezve**
This PR is a revived form of PR #63. It introduces the ``exit_limit`` as a way to specify the maximum number of seats in the cabin. This is stored as a field in ``fuselage.cabin``. In addition, the name of the input ``max_pax`` is changed to the more accurate ``max_payload_in_pax_equivalent``.

The code now uses the exit limit to resize the fuselage where appropriate. The PR also fixes a very minor bug in the double-decker case (a ``parg`` parameter had been left behind).

### CODE REVIEW

File: src/structures/update_fuse.jl [hunk 1]
```diff
@@ -183,7 +182,13 @@ function find_minimum_radius_for_seats_per_row(seats_per_row, ac_base)
     (minf,xopt,ret) = NLopt.optimize(opt, xopt) #Solve optimization problem starting from global solution
 
     R = xopt[1]
-    return R
+    #Check if constraint is met
+    diff = check_seats_per_row_diff(seats_per_row, xopt, ac)
```
Comment (argonaut22) [inline src/structures/update_fuse.jl:186]: how is `return 0.0` handled upstream? 

on that note, this doesn't seem to be used anywhere rn

Comment (ngomezve) [reply src/structures/update_fuse.jl:186]: This function is not used in sizing. It is used outside of sizing if you want to update the fuselage diameter, for example, if you have a hydrogen tank.

Comment (ngomezve) [reply src/structures/update_fuse.jl:186]: The `return 0.0` is a failure case. I will replace it with an error so that it is clearer.

**Comment by argonaut22** [comment]
@ngomezve, just finished reviewing. left brief notes in the `[Merge branch 'main' into dev_cargo_ref](https://github.com/MIT-LAE/TASOPT.jl/pull/81/commits/2e87f30b3dd5d8d45efcef4086f519640a755656)` commit instead of this thread by accident 🤦. overall, looks good

**Comment by ngomezve** [comment]
Thanks @argonaut22. Unfortunately, it seems like you reviewed the diffs in a merge commit from main, instead of the overall diffs in the PR! In essence, commit (https://github.com/MIT-LAE/TASOPT.jl/commit/2e87f30b3dd5d8d45efcef4086f519640a755656) contains diffs from older PRs already in TASOPT that I was merging into this branch to resolve merge conflicts. Could you please review the actual changes in this PR? You can do it by going to the "Files changed tab". It should be very short as I didn't add a lot of code.

**Review by argonaut22 — APPROVED** [review]
Reviewed and approved. No issues with these edits.

Some notes from my extraneous deep dive into prior commits:
- @ngomezve, on the [obj fxn errors](https://github.com/MIT-LAE/TASOPT.jl/commit/2e87f30b3dd5d8d45efcef4086f519640a755656#r148111088), we do wanna make any errors as transparent as possible but i'll trust your judgment.
- as for [`maxpax`/`max_payload_in_pax_equivalent`](https://github.com/MIT-LAE/TASOPT.jl/commit/2e87f30b3dd5d8d45efcef4086f519640a755656#r148068270) and [flag variable conventions](https://github.com/MIT-LAE/TASOPT.jl/commit/2e87f30b3dd5d8d45efcef4086f519640a755656#r148111337), I'm happy to make the tweaks on a separate PR. Any objections @askprash?

**Comment by askprash** [comment]
A separate PR for that sounds good. 👍

**Pull request merged**